### PR TITLE
[PLT-6710] /rename [text] slash command: Rename the channel

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
/rename [text] slash command: Rename the channel

#### Ticket Link
Jira: https://mattermost.atlassian.net/browse/PLT-6710
GitHub: https://github.com/mattermost/platform/issues/6558

#### Checklist
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

**Comment by jasonblais** [comment]
@cpanato Now that #6566 is merged, we can apply the same changes to this PR?

**Comment by cpanato** [comment]
@jasonblais I will wait for this https://github.com/mattermost/platform/pull/6569 to be merged

**Comment by jasonblais** [comment]
Looks like it got merged now :)

**Comment by cpanato** [comment]
@jasonblais ready for a new round :)

File: i18n/en.json [hunk 1]
```diff
@@ -604,6 +604,42 @@
...
+    "id": "api.command_channel_rename.desc",
+    "translation": "Rename the channel."
+  },
+  {
+    "id": "api.command_channel_rename.channel.app_error",
+    "translation": "Error to retrieve the current channel."
+  },
+  {
+    "id": "api.command_channel_rename.permission.app_error",
+    "translation": "You do not have the appropriate permissions to edit the channel purpose."
```
Comment (jasonblais) [inline i18n/en.json]: @cpanato I think this should be `You do not have the appropriate permissions to rename the channel.`

File: i18n/en.json [hunk 2]
```diff
@@ -604,6 +604,42 @@
...
+    "id": "api.command_channel_rename.name",
+    "translation": "rename"
+  },
+  {
+    "id": "api.command_channel_rename.hint",
+    "translation": "[text]"
+  },
+  {
+    "id": "api.command_channel_rename.desc",
+    "translation": "Rename the channel."
```
Comment (jasonblais) [inline i18n/en.json]: Remove period so it's consistent with other autocomplete descriptions

`Rename the channel`

File: i18n/en.json [hunk 3]
```diff
@@ -604,6 +604,42 @@
...
+    "id": "api.command_channel_rename.update_channel.app_error",
+    "translation": "Error to update the current channel."
+  },
+  {
+    "id": "api.command_channel_rename.update_channel.success",
+    "translation": "Channel name successfully updated."
+  },
+  {
+    "id": "api.command_channel_rename.direct_group.app_error",
+    "translation": "Cannot rename the Channel for direct message channels. Use /header to set the header instead."
```
Comment (jasonblais) [inline i18n/en.json]: Propose `Cannot rename direct message channels.`

File: i18n/en.json (hunk 1 above)
Comment (cpanato) [inline i18n/en.json]: done

File: i18n/en.json (hunk 2 above)
Comment (cpanato) [inline i18n/en.json]: done

File: i18n/en.json (hunk 3 above)
Comment (cpanato) [inline i18n/en.json]: done

**Comment by cpanato** [comment]
@jasonblais ready for you again :)

**Pull request merged**